spec/02-integration/05-proxy/10-balancer/03-consistent-hashing_spec.lua

```diff
@@ -215,11 +215,19 @@ for _, strategy in helpers.each_strategy() do
           bu.end_testcase_setup(strategy, bp)
 
           local client = helpers.proxy_client()
-          local res = assert(client:request({
-            method = "GET",
-            path = uri,
-            headers = { host = api_host },
-          }))
+
+          local res
+          helpers.wait_until(function()
```

Comment:
Seeing this pattern added in so many integration tests is concerning to me, because:

* `helpers.wait_until` is generic and doesn't really provide meaningful output when it fails, so it's harder to debug failing tests
* it makes me worry that there are many more test cases which may be flaky, and we could be chasing after them for a long time


Is there no alternative to this? Is there any function within the new events library to block until all workers have handled/seen events up until some known event? If so, we could expose it via the admin API and incorporate it into `spec.helpers`. Alternatively, we could add an optional, (maybe private) query arg to the CRUD APIs for traditional mode, so that we can do things like `POST /services?sync=true`, and it would block until all workers have caught up with events.

Additionally, something that might help would be to set `nginx_main_worker_processes` to 1 in most cases, since this will cut down on how much time is spent ensuring all workers are in sync.

Reply:
The new events library affects too many test cases, It is a hard work to make all of them green.
I think if we could improve these test cases after this PR is merged.